content/docs/state-and-lifecycle.md

```diff
@@ -409,41 +401,41 @@ Then you can update them independently with separate `setState()` calls:
   }
 ```
 
-The merging is shallow, so `this.setState({comments})` leaves `this.state.posts` intact, but completely replaces `this.state.comments`.
+Միավորումը մակերեսային է, այսինքն\` `this.setState({comments})`-ը թողնում է `this.state.posts`-ին «անվնաս», բայց ամբողջությամբ փոխարինում է `this.state.comments`-ը։
 
-## The Data Flows Down {#the-data-flows-down}
+## Տվյալների հոսքը ներքև {#the-data-flows-down}
 
-Neither parent nor child components can know if a certain component is stateful or stateless, and they shouldn't care whether it is defined as a function or a class.
+Ո'չ ծնող, ո'չ զավակ կոմպոնենտները չեն կարող իմանալ, արդյոք տվյալ կոմպոնենտը վիճակով(stateful) է կամ վիճակազուրկ(stateless), և նրանք չպետք է հոգ տանեն այն բանի մասին, թե դա հայտարարված է որպես ֆունկցիա, թե որպես կլաս։
 
-This is why state is often called local or encapsulated. It is not accessible to any component other than the one that owns and sets it.
+Ահա թե ինչու state-ը հաճախ անվանվում է լոկալ կամ ինկապսուլացված։ Այն հասանելի չէ ուրիշ ոչ մի կոմպոնենտից, բացի այն մեկից, որին պատկանում է և որը տեղադրել է իրեն։
 
-A component may choose to pass its state down as props to its child components:
+Կոմպոնենտը կարող է փոխանցել իր state-ը ներքև որպես props, իր զավակ կոմպոնենտներին.
 
 ```js
-<h2>It is {this.state.date.toLocaleTimeString()}.</h2>
+<h2>Ժամը {this.state.date.toLocaleTimeString()}-ն է։</h2>
 ```
 
-This also works for user-defined components:
+Սա նաև աշխատում է օգտագործողի կողմից հայտարարված կոմպոնենտների համար.
 
 ```js
 <FormattedDate date={this.state.date} />
 ```
 
-The `FormattedDate` component would receive the `date` in its props and wouldn't know whether it came from the `Clock`'s state, from the `Clock`'s props, or was typed by hand:
+`FormattedDate`-ը կստանա `date`-ը իր props-ում և չի իմանա այն եկել էր `Clock`-ի state-ից, `Clock`-ի props-ից, թե գրված էր ձեռքով.
 
 ```js
 function FormattedDate(props) {
-  return <h2>It is {props.date.toLocaleTimeString()}.</h2>;
+  return <h2>Ժամը {props.date.toLocaleTimeString()}-ն է։</h2>;
 }
 ```
 
-[**Try it on CodePen**](https://codepen.io/gaearon/pen/zKRqNB?editors=0010)
+[**Փորձել CodePen-ում**](https://codepen.io/gaearon/pen/zKRqNB?editors=0010)
 
-This is commonly called a "top-down" or "unidirectional" data flow. Any state is always owned by some specific component, and any data or UI derived from that state can only affect components "below" them in the tree.
+Սա սովորաբար կոչվում է «վերևից-ներքև» կամ «միակողմանի» տվյալների հոսք։ Ցանկացած state միշտ պատկանում է որևէ սպեցիֆիկ կոմպոնենտի, և ցանկացած տվյալ կամ UI ստացված state-ից կարող է միայն ազդել ծառում «ներքև» գտնվող կոմպոնենտների վրա։
```

Comment:
"derived from that state"֊> "ստացված ԱՅԴ state-ից"
"կարող է միայն ազդել ծառում «ներքև» գտնվող կոմպոնենտների վրա" ֊> "կարող է ԱԶԴԵԼ ՄԻԱՅՆ ծառում ԴՐԱՆՑԻՑ «ներքև» գտնվող կոմպոնենտների վրա"